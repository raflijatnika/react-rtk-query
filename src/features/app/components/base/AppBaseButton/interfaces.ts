// Antd
import { ButtonProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseButtonProps extends ButtonProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
  width?: number
  height?: number
}
