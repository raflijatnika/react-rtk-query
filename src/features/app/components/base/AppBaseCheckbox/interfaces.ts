// Antd
import { CheckboxProps } from 'antd'

export type IAppBaseCheckboxProps = CheckboxProps
