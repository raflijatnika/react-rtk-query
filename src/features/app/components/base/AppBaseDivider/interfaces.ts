// Antd
import { DividerProps } from 'antd'

export type IAppBaseDividerProps = DividerProps
