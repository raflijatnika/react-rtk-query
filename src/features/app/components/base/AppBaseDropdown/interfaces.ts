// React
import { ReactElement } from 'react'

// Interfaces
import { DropdownProps, MenuProps } from 'antd'

export interface IDropdownProps {
  children?: ReactElement
  trigger?: DropdownProps['trigger']
  items: MenuProps['items']
}
