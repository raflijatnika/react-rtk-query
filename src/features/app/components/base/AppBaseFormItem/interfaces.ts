// Interfaces
import { FormItemProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseFormItemProps extends FormItemProps {
  labelColor?: APP_COLOR | APP_COLOR_LIGHT
  labelSize?: number
}
