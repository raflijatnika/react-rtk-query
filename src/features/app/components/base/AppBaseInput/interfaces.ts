// Interfaces
import { InputProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseInputProps extends InputProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  height?: number
  width?: number
}
