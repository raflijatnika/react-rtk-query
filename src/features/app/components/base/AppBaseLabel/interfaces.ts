// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseLabelProps {
  fontWeight?: number
  fontSize?: number
  isBold?: boolean
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
}
