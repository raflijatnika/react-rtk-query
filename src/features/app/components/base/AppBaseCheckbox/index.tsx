// Styled Components
import styled from 'styled-components'

// Antd
import { Checkbox } from 'antd'

// Interfaces
import { IAppBaseCheckboxProps } from './interfaces'

const AppBaseCheckbox = styled(Checkbox).attrs(
  (props): IAppBaseCheckboxProps => props
)<IAppBaseCheckboxProps>``

export { AppBaseCheckbox }
