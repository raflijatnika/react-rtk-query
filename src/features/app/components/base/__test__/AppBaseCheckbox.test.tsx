// Test
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { ReactTestRenderer } from 'react-test-renderer'
import 'jest-styled-components'

// Components
import { AppBaseCheckbox } from '../AppBaseCheckbox'

// Interfaces
import { IAppBaseCheckboxProps } from '../AppBaseCheckbox/interfaces'

const rendererInstance = (props?: IAppBaseCheckboxProps): ReactTestRenderer =>
  renderer.create(<AppBaseCheckbox {...props} />)

it('AppBaseCheckbox should render', () => {
  render(<AppBaseCheckbox />)
})

it('AppBaseCheckbox should have snapshot', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toMatchSnapshot()
})
