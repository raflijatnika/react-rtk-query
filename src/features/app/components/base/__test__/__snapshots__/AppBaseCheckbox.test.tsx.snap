// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBaseCheckbox should have snapshot 1`] = `
<label
  className="ant-checkbox-wrapper "
>
  <span
    className="ant-checkbox"
    style={Object {}}
  >
    <input
      checked={false}
      className="ant-checkbox-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      type="checkbox"
    />
    <span
      className="ant-checkbox-inner"
    />
  </span>
</label>
`;
